### PR TITLE
Remove obsolete instructions from scenario_app readme, pass depfile correctly.

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/90303 by removing out of date instructions. @blasten, I filed https://github.com/flutter/flutter/issues/90401 but please feel free to dedupe if there is an existing issue for that already.

Fixes https://github.com/flutter/flutter/issues/88179 by shifting the location of the `--depfile` argument to be to the frontend server instead of to the VM. This does have the drawback that it will fail to recompile properly if the frontend server sources are modified, but unfortunately I can't see a way to get a single combined depfile for all the sources.

### CODE REVIEW

File: testing/scenario_app/README.md [hunk 1]
```diff
@@ -21,10 +21,14 @@ platform channel.
...
 ## Running for iOS
 
+Build the `ios_debug_sim_unopt` engine variant, and run
+
 ```sh
-./build_and_run_ios_tests.sh
+./run_ios_tests.sh
 ```
 
+in your shell.
```
Comment (cyanglaz) [inline testing/scenario_app/README.md:30]: While you are here, do you mind adding a little instruction for running locally through xcode?
Something like:
```
To run or debug in XCode, open the proj file located in  <path_to_engine>/src/out/ios_debug_sim_unopt/scenario_app/Scenarios/Scenarios.xcodeproj, and run all tests with cmd+u or use XCode UI to run selected tests.
```

**Review by cyanglaz — APPROVED** [review]
LGTM

File: testing/scenario_app/README.md (hunk 1 above)
Comment (dnfield) [reply testing/scenario_app/README.md:30]: Will do

Comment (cyanglaz) [reply testing/scenario_app/README.md:30]: thanks!

**Comment by chinmaygarde** [comment]
Dan is going to rebase on ToT. The ci.yaml validation failure is unrelated and is probably resolved.

**Pull request merged**